### PR TITLE
HTBHF-637 Initial commit. Added models and endpoint.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+    
+* The first numeric digit represents the number of children under one.
+  e.g BA200000C will return a valid response with two children under one.
+  
+* The second numeric digit represents the number of children under four.
+  e.g BA030000C will return a valid response with three children under four.
+  
+* The third numeric digit represents if the person is on universal credits. Any digit accept '0' means the person is receiving universal credits.
+
+* The fourth numeric digit represents if the person is on child tax credits. Any digit accept '0' means the person is receiving child tax credits.
```
Comment (YetAnotherMatt) [inline README.md]: Universal credit and CTC are mutually exclusive, and come from different APIs. DWP api will return UC, HMRC api will return CTC

File: .gitignore [hunk 2]
```diff
@@ -10,5 +10,4 @@ gradle-app.setting
 # Cache of project
 .gradletasknamecache
 
-# # Work around https://youtrack.jetbrains.com/issue/IDEA-116898
-# gradle/wrapper/gradle-wrapper.properties
+.idea/
```
Comment (YetAnotherMatt) [inline .gitignore]: we probably want to add /out here, and delete all the class files...

File: README.md (hunk 1 above)
Comment (dwybourn) [reply README.md]: Ok, I've updated such that the last character represents the benefit.

File: .gitignore (hunk 2 above)
Comment (dwybourn) [reply .gitignore]: Done.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,22 @@
...
+  e.g. AB123456C will return 404 not found, 
+     DB123456C will return a valid response.
+    
+* The first numeric digit represents the number of children under one.
+  e.g BA200000C will return a valid response with two children under one.
+  
+* The second numeric digit represents the number of children under four.
+  e.g BA030000C will return a valid response with three children under four.
+  
+* The last character represents the benefits a person.
```
Comment (mikeSmith08) [inline README.md]: You're missing a word or two here.

File: src/main/java/uk/gov/dhsc/htbhf/smartstub/service/BenefitsService.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        var childrenUnderFour = getNumberOfChildrenUnderFour(nino);
+
+        return BenefitDTO.builder()
+                .benefit(benefit)
+                .numberOfChildrenUnderOne(childrenUnderOne)
+                .numberOfChildrenUnderFour(childrenUnderFour)
+                .build();
+    }
+
+    private boolean userNotFound(char[] nino) {
```
Comment (mikeSmith08) [inline src/main/java/uk/gov/dhsc/htbhf/smartstub/service/BenefitsService.java]: I'd call this `personNotFound` to match the constants and Exception.

File: README.md (hunk 3 above)
Comment (dwybourn) [reply README.md]: Fixed.

File: src/main/java/uk/gov/dhsc/htbhf/smartstub/service/BenefitsService.java (hunk 4 above)
Comment (dwybourn) [reply src/main/java/uk/gov/dhsc/htbhf/smartstub/service/BenefitsService.java]: Done.

File: src/main/java/uk/gov/dhsc/htbhf/smartstub/controller/DWPBenefitController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    private BenefitsService benefitsService;
+
+    public DWPBenefitController(BenefitsService benefitsService) {
+        this.benefitsService = benefitsService;
+    }
+
+    @PostMapping
+    public BenefitDTO getBenefits(@RequestBody PersonDTO person) {
+        return benefitsService.getBenefits(person.getNino().toCharArray());
```
[the remark this answers is not in the document]
Comment (dwybourn) [reply src/main/java/uk/gov/dhsc/htbhf/smartstub/controller/DWPBenefitController.java]: Personally I think nino[4] reads better than nino.charAt(4)

**Pull request merged**